hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/keygen/TestCustomKeyGenerator.java

```diff
@@ -98,53 +103,145 @@ private TypedProperties getPropertiesForNonPartitionedKeyGen() {
   }
 
   @Test
-  public void testSimpleKeyGenerator() {
-    BuiltinKeyGenerator keyGenerator = new CustomKeyGenerator(getPropertiesForSimpleKeyGen());
+  public void testSimpleKeyGenerator() throws IOException {
+    TypedProperties propertiesForSimpleKeyGen = getPropertiesForSimpleKeyGen();
+
+    // Test config with HoodieWriteConfig.KEYGENERATOR_CLASS_PROP
+    propertiesForSimpleKeyGen.put(HoodieWriteConfig.KEYGENERATOR_CLASS_PROP, CustomKeyGenerator.class.getName());
+
+    BuiltinKeyGenerator keyGenerator = HoodieSparkKeyGeneratorFactory.createKeyGenerator(propertiesForSimpleKeyGen);
     GenericRecord record = getRecord();
     HoodieKey key = keyGenerator.getKey(record);
     Assertions.assertEquals(key.getRecordKey(), "key1");
     Assertions.assertEquals(key.getPartitionPath(), "timestamp=4357686");
     Row row = KeyGeneratorTestUtilities.getRow(record);
     Assertions.assertEquals(keyGenerator.getRecordKey(row), "key1");
     Assertions.assertEquals(keyGenerator.getPartitionPath(row), "timestamp=4357686");
+
+    // Test config with HoodieWriteConfig.KEYGENERATOR_TYPE_PROP
+    propertiesForSimpleKeyGen = getPropertiesForSimpleKeyGen();
+    propertiesForSimpleKeyGen.put(HoodieWriteConfig.KEYGENERATOR_TYPE_PROP, KeyGeneratorType.CUSTOM.name());
+
+    keyGenerator = HoodieSparkKeyGeneratorFactory.createKeyGenerator(propertiesForSimpleKeyGen);
```

Comment:
we could trying moving this repetitive code to a common private method and re-use them. I mean lines 125 to 132. Similarly for all key types.

Reply:
or better option would be to go with Parameterized tests.